src/Features/Core/Portable/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs

```diff
@@ -0,0 +1,94 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.UseCompoundAssignment
+{
+    internal abstract class AbstractUseCompoundAssignmentCodeFixProvider<
+        TSyntaxKind, TAssignmentSyntax, TExpressionSyntax> 
+        : SyntaxEditorBasedCodeFixProvider
+        where TSyntaxKind : struct
+        where TAssignmentSyntax : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseCompoundAssignmentDiagnosticId);
+
+        // See comments in the analyzer for what these maps are for.
+
+        private readonly ImmutableDictionary<TSyntaxKind, TSyntaxKind> _binaryToAssignmentMap;
+        private readonly ImmutableDictionary<TSyntaxKind, TSyntaxKind> _assignmentToTokenMap;
+
+        protected AbstractUseCompoundAssignmentCodeFixProvider(
+            ImmutableArray<(TSyntaxKind exprKind, TSyntaxKind assignmentKind, TSyntaxKind tokenKind)> kinds)
+        {
+            Utilities.GenerateMaps(kinds, out _binaryToAssignmentMap, out _assignmentToTokenMap);
+        }
+
+        protected abstract TSyntaxKind GetSyntaxKind(int rawKind);
+        protected abstract SyntaxToken Token(TSyntaxKind kind);
+        protected abstract TAssignmentSyntax Assignment(
+            TSyntaxKind assignmentOpKind, TExpressionSyntax left, SyntaxToken syntaxToken, TExpressionSyntax right);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics[0];
+            
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(document, diagnostic, c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var assignment = diagnostic.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+                editor.ReplaceNode(assignment,
+                    (currentAssignment, generator) =>
+                    {
+                        syntaxFacts.GetPartsOfAssignmentExpressionOrStatement(currentAssignment, 
+                            out var leftOfAssign, out var equalsToken, out var rightOfAssign);
+
+                        syntaxFacts.GetPartsOfBinaryExpression(rightOfAssign,
+                           out _, out var opToken, out var rightExpr);
```

Comment:
where trivia around opToken go?

Reply:
nowhere.  It does not seem like a particular common coding pattern to do `x = x + /*comment*/ y`.  So we do not do anything special here.  This fits under rule '1' from the discussion above.

Reply:
As a data point, this feature found more than 100 places in roslyn that could be fixed up.  However, none of them had trivia there.